src/components/Modal.vue

```diff
@@ -0,0 +1,217 @@
+              <p class="release-date">{{ movie.release_date.split('-')[0] }}</p>
+              </h2>
+              <p class="movie-rating">{{ movie.user_rating }}</p>
+              </div>
+            <article class="overview-box">
+              <i>
+                {{ movie.overview }}
+              </i>
+            </article>
+            <aside class="button-box">
+              <p name="add" v-on:click="emitWatchlist" v-if="list !== 'My Watchlist'">Add to Watchlist</p>
+              <p v-else name="remove" v-on:click="emitWatchlist">Remove from Watchlist</p>
+              <button class="thumb" name="upVote" v-on:click="emitThumbs">👍</button>
+              <button class="thumb" name="downVote" v-on:click="emitThumbs">👎</button>
+            </aside>
+          </section>
+      </div>
+    </div>
+  </transition>
+</template>
+
+<script>
+export default {
+  name: "Modal",
+  data() {
+    return {
+      show: false
+    };
+  },
+  props: {
+    movie: Object,
+    showing: Boolean
+  },
+  methods: {
+    closeModal() {
+      this.$emit('closeModal')
+      document.querySelector("body").classList.remove("overflow-hidden");
+    },
+    openModal() {
+      console.log(this.movie.backdrop)
+      this.$emit('openModal')
+      document.querySelector("body").classList.add("overflow-hidden");
+    },
+    emitThumbs(e) {
+      if (e.target.name === 'upVote') {
+        this.$emit('postThumb', {api_movie_id: this.movie.id, title: this.movie.title, up: true});
+      } else {
+        this.$emit('postThumb', {api_movie_id: this.movie.id, title: this.movie.title, up: false});
+      }
+    },
+    emitWatchlist() {
+      this.$emit('postWatchList', {api_movie_id: this.movie.id, title: this.movie.title})
+    },
+  }
+};
+</script>
+
+
+<style lang="scss" scoped>
+@import "../index.scss";
+.modal {
+  overflow-x: hidden;
+  overflow-y: auto;
+  position: fixed;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  z-index: 9;
+  &__backdrop {
+    background-color: rgba(0, 0, 0, 0.3);
+    position: fixed;
+    top: 0;
+    right: 0;
+    bottom: 0;
+    left: 0;
+    z-index: 1;
+  }
+  &__dialog {
+    background-color: #ffffff;
+    position: relative;
+    width: 90%;
+    height: 90%;
+    margin: 50px auto;
+    display: flex;
+    border-radius: 5px;
+    z-index: 2;
+    padding: 50px;
+    overflow-y: hidden;
+    svg {
+      color: $gray;
+      cursor: pointer
+    }
+    @media screen and (max-width: 992px) {
+      width: 90%;
+    }
+  }
+  &__close {
+    width: 30px;
+    height: 30px;
+  }
+  &__header {
+    padding: 20px 20px 10px;
+    display: flex;
+    align-items: flex-start;
+    justify-content: space-between;
+  }
+  &__body {
+    padding: 10px 20px 10px;
+    overflow: auto;
+    display: flex;
+    flex-direction: column;
+    align-items: stretch;
+  }
+  &__footer {
+    padding: 10px 20px 20px;
+  }
+}
+.fade-enter-active,
+.fade-leave-active {
+  transition: opacity 0.2s;
```

Comment:
This is great! I wonder if we can use more of these transitions in other places.